tests/saitests/py3/sai_qos_tests.py

```diff
@@ -4024,91 +4041,72 @@ def runTest(self):
+                    try:
+                        if (recv_pkt.payload.src == src_port_ip) and (recv_pkt.payload.dst == dst_port_ip):
+                            total_recv_cnt += 1
+                            pkts.append(recv_pkt)
+                    except AttributeError:
+                        continue
+                    except IndexError:
+                        # Ignore captured non-IP packet
+                        continue
+                print("Total packet recieved: " + str(total_recv_cnt))
                 for pkt_to_inspect in pkts:
-                    pkt_str = hex_dump_buffer(pkt_to_inspect)
-
                     # Count marked and not marked amount of packets
-                    if ((int(pkt_str[ECN_INDEX_IN_HEADER]) & 0x03) == 1):
+                    if ((pkt_to_inspect.payload.tos & 0x03) == 1) or ((pkt_to_inspect.payload.tos & 0x03) == 2):
                         not_marked_cnt += 1
-                    elif ((int(pkt_str[ECN_INDEX_IN_HEADER]) & 0x03) == 3):
-                        assert (not_marked_cnt == 0)
+                    elif ((pkt_to_inspect.payload.tos & 0x03) == 3):
                         marked_cnt += 1
 
                 print("    ECN non-marked pkts: " + str(not_marked_cnt))
                 print("    ECN marked pkts:     " + str(marked_cnt))
                 print("")
 
-            time.sleep(5)
-            # Read Counters
-            print("DST port counters: ")
-            port_counters, queue_counters = sai_thrift_read_port_counters(
-                self.dst_client, asic_type, port_list['dst'][dst_port_id])
-            print(port_counters)
-            print(queue_counters)
+            dut_port = self.get_dut_port(dst_port_id)
+            if platform_asic and platform_asic == "broadcom-dnx":
+                time.sleep(8)
+                stdOut, stdErr, retValue = self.exec_cmd_on_dut(self.dst_server_ip, self.test_params['dut_username'],
+                                                                self.test_params['dut_password'],
+                                                                'show queue wredcounters -n asic{}| grep {}| grep UC{}'
+                                                                .format(self.dst_asic_index, dut_port, dscp))
+                if stdErr and retValue != 0:
+                    raise RuntimeError("Command might have failed in the DUT.Error:{}".format(stdErr))
+                else:
+                    print("----Queue WredCounters on DUT----")
+                    print("ECNMarked pkts: {}  ECNMarked bytes: {}".format(stdOut[0].split()[4],
+                                                                           stdOut[0].split()[5]))
+                # the output number format is comma seperated value,remove comma if expected marked_pkts > 999
+                if ecn_queue_status == 'off':
+                    assert (marked_cnt == 0)
+                assert (int(stdOut[0].split()[4]) == marked_cnt)
```

Comment:
It seems we are verifying marked packet count using show command to make sure it matches. Shall we check this with expected marked packet count as well ?